### ISSUE TITLE
Infinite event is not fired since 2.2.2

### ISSUE THREAD

**Issue by rquelen** [PeachScript/vue-infinite-loading#111]
### Version
2.2.2

### Vue.js version
2.4.2 (with nuxt 1.0.0-rc11)

### What is expected?
infinite-loading should fire 'infinite' event as usual

### What is actually happening?
the event is not fired anymore since 2.2.2 (working in 2.2.1)

### How to reproduce this problem?
Use the component as usual, but the handler is never called:
```
<no-ssr>
  <InfiniteLoading @infinite="infiniteHandler"></InfiniteLoading>
</no-ssr>
...
methods: {
  infiniteHandler ($state) { console.log('never called') },
},
```

**Comment by PeachScript** [comment]
Did you used the `vue-no-ssr` component? I tried it like your code, and it works for me, this is a live demo: https://jsfiddle.net/ksLr54sw/

**Comment by rquelen** [comment]
Yes we did use `vue-no-ssr` (from nuxt), and our code worked with 2.2.1 so we rollbacked to this version for the moment.
Your jsfiddle seems to work, but it doesn't use nuxt and is not run in Server Side Rendering.
So maybe the difference comes from here ? I don't know how to make a simple jsfiddle with nuxt...

**Comment by filipesarturi** [comment]
Same problem here

**Comment by pirmax** [comment]
Same problem, sometimes, the function isn't fired...

**Comment by filipesarturi** [comment]
Here it always happens in version 2.2.2, back to version 2.2.1 works normally.

**Comment by PeachScript** [comment]
@filipesarturi @pirmax did you also used it with `nuxt`?

**Comment by filipesarturi** [comment]
@PeachScript no

**Comment by roosh-t3** [comment]
Event is not firing, using plain VueJS

**Comment by PeachScript** [comment]
@filipesarturi @pirmax @roosh-t3 could you reproduce this problem through a live demo like this https://jsfiddle.net/ksLr54sw/ to help me find the reason?

**Comment by kitkimwong** [comment]
Same problem, total can't use

**Comment by damienbeaufils** [comment]
Hello,

I have the same problem, and I took the time to build my application with each commit between 2.2.1 and 2.2.2 to find which one lead to this issue.

I found that the commit 8b47b2d which fixes #100 is causing the actual bug. It is because when the `elm` is `body`, it can pass [the first `if`](https://github.com/PeachScript/vue-infinite-loading/blob/master/src/components/InfiniteLoading.vue#L273) or [the second `if`](https://github.com/PeachScript/vue-infinite-loading/blob/master/src/components/InfiniteLoading.vue#L275)  (depending on parameters and element style). But not matter which `if` is ran, when the found element is `body`, the result should always be `window` instead of `body`.

@PeachScript I've made a PR to fix this issue (and I hope keep the other issue #100 fixed too): #115

**Comment by PeachScript** [comment]
@damienbeaufils thank you very much to spend time to find the reason! After reading your comment,  I think I made a mistake when fix the #100 , before that the program will find the body first, code is [here](https://github.com/PeachScript/vue-infinite-loading/blob/2f2a2352784266cb6e5ebf7794dd21c526631776/src/components/InfiniteLoading.vue#L266), I think it is the same with your PR, so is the #100 a special situation? Maybe I should recommend the reporter of #100 to use the `force-use-infinite-wrapper` attribute?

Thanks again!

**Comment by damienbeaufils** [comment]
@PeachScript I agree that #100 is a special case and should use the `force-use-infinite-wrapper` instead

**Comment by PeachScript** [comment]
@damienbeaufils I see, I have been released a new version [v2.2.3](https://github.com/PeachScript/vue-infinite-loading/releases/tag/v2.2.3) to fix it, you can try it now, and thanks for your idea and PR :D

**Comment by PeachScript** [comment]
@rquelen @filipesarturi @pirmax @kitkimwong @roosh-t3 hello everyone, you can try the new version [v2.2.3](https://github.com/PeachScript/vue-infinite-loading/releases/tag/v2.2.3) now, if it still not working, please tell me, thanks!
